nuxeo-features/nuxeo-annotation/nuxeo-annotation-core/src/main/java/org/nuxeo/ecm/annotation/AnnotationImpl.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Funsho David
+ *
+ */
+
+package org.nuxeo.ecm.annotation;
+
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_COLOR_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_CREATION_DATE_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_DATE_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_DOCUMENT_ID_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_DOCUMENT_XPATH_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_FLAGS_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_LAST_MODIFIER_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_NAME_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_OPACITY_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_PAGE_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_POSITION_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_SECURITY_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_SUBJECT_PROPERTY;
+
+import java.io.Serializable;
+import java.util.Calendar;
+import java.util.Map;
+
+import org.nuxeo.ecm.core.api.DocumentModel;
+
+/**
+ * @since 10.1
+ */
+public class AnnotationImpl implements Annotation {
+
+    protected String id;
+
+    protected String color;
+
+    protected Calendar date;
+
+    protected String flags;
+
+    protected String name;
+
+    protected String documentId;
+
+    protected String documentXpath;
+
+    protected String lastModifier;
+
+    protected long page;
+
+    protected String position;
+
+    protected Calendar creationDate;
+
+    protected double opacity;
+
+    protected String subject;
+
+    protected String security;
+
+    public AnnotationImpl() {
+    }
+
+    public AnnotationImpl(DocumentModel annotationModel) {
```

Comment:
I'd prefer to make the construction from `DocumentModel` protected in the package. External callers have no reason to do it.

Reply:
As `AnnotationImpl` is the reference POJO for the `Annotation` interface I think it's worth putting it in the annotation-api module.